common/src/main/java/com/microsoft/identity/common/internal/broker/BrokerValidator.java

```diff
@@ -257,7 +256,24 @@ private X509Certificate getSelfSignedCert(final List<X509Certificate> certs)
     public static boolean isValidBrokerRedirect(@Nullable final String redirectUri,
                                                 @NonNull final Context context,
                                                 @NonNull final String packageName) {
-        return StringUtil.equalsIgnoreCase(redirectUri, getBrokerRedirectUri(context, packageName));
+        final String methodName = ":isValidBrokerRedirect";
+        final String expectedBrokerRedirectUri = getBrokerRedirectUri(context, packageName);
+        final boolean isValidBrokerRedirect = StringUtil.equalsIgnoreCase(redirectUri, expectedBrokerRedirectUri);
+
+        if (!isValidBrokerRedirect) {
+            Logger.error(
+                    TAG + methodName,
+                    "Broker redirect uri is invalid."
+                            + "\n"
+                            + "Expected: " + expectedBrokerRedirectUri
+                            + "\n"
+                            + "Actual: " + redirectUri
+                    ,
+                    null
```

Comment:
nit: Do we need a multiline log statement here?

Reply:
_Personally_ I think it's easier to read, but I appreciate this is subjective. Is your suggestion to condense this to a single line or to log this altogether differently?

Reply:
A single line is best.  If we need to put structure on it, we can do it inside of a line.  (Log injection attacks are a thing.)

Reply:
d64813b138042f882c398ad2899edf21019720f3